### ISSUE TITLE
Get ready for ES modules in the new PureScript version

### ISSUE THREAD

**Issue by i-am-the-slime** [MaybeJustJames/zephyr#67]
Hey @coot 

Happy new year 🥳 

I think it'd be great if zephyr worked with the new version of the compiler when it gets released.

I've tried to incorporate some older changes from @kl0tl but something seems amiss.

It'd be great if you could have a look at the changes here: https://github.com/working-group-purescript-es/zephyr/pull/1/files

**Comment by MaybeJustJames** [comment]
@coot is planning to step away from `zephyr` development (https://discord.com/channels/864614189094928394/928230092252004352/928235541370122280)
I've volunteered myself to try to integrate zephyr dce into the compiler. Does it make sense that I work from https://github.com/working-group-purescript-es/purescript or https://github.com/purescript/purescript ?

**Comment by MaybeJustJames** [comment]
This is an error from node:
```
$ node --version
v16.13.0

$ node index.js 
(node:3784672) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
test/lib-tests/dce-output/Literals/index.js:21
export {
^^^^^^

SyntaxError: Unexpected token 'export'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1031:15)
    at Module._compile (node:internal/modules/cjs/loader:1065:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
```
Indeed, after a manual rename of the file the test passes successfully. I'm not in a position to assess what the right course of action is. Should the output become `*.mjs` for ES modules?

**Comment by i-am-the-slime** [comment]
Which file do you mean? I think `run.js` can just be a `run.mjs`. My main concern is the lookup failure in the linked ticket.

**Comment by thomashoneyman** [comment]
> Should the output become *.mjs for ES modules?

I believe the compiler output directory includes a `package.json` file that sets `type: "module"` so that things run properly on Node

**Comment by toastal** [comment]
I didn't get an immediate answer on Matrix. Was this merged into the compiler or not? Is it/will it be a recommended tool for v0.15.x PureScript code? The current fate seems a bit unclear...

**Comment by MaybeJustJames** [comment]
Hi @toastal,
I think the long term goal is to merge into the compiler. I'm not currently working on that but I intend to if noone picks it up. I need to do a release for 0.15 (exists in a branch at the moment). Until the compiler merge happens I do intend to maintain zephyr as a usable separate tool. That said I haven't done any measurements to check if it's useful vs the DCE that bundlers already do.

**Comment by MaybeJustJames** [comment]
Closing because I released 0.5 which uses PureScript 0.15 for es modules support
